Packages/com.unity.multiplayer.samples.coop/Utilities/SceneManagement/SceneLoader.cs

```diff
@@ -0,0 +1,141 @@
+using System.Collections.Generic;
+using Unity.Netcode;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace Unity.Multiplayer.Samples.Utilities
+{
+    public class SceneLoader : NetworkBehaviour
+    {
+        /// <summary>
+        /// This singleton is a utility handling the loading of scenes. It loads scene using the SceneManager, or, on
+        /// listening servers for which scene management is enabled, using the NetworkSceneManager. It also starts and
+        /// stops the loading screen.
+        /// </summary>
+
+        [SerializeField]
+        ClientLoadingScreen m_ClientLoadingScreen;
+
+        public static SceneLoader Instance { get; private set; }
+
+        public void Awake()
+        {
+            if (Instance != null && Instance != this)
+            {
+                Destroy(this.gameObject);
+            }
+            else
+            {
+                Instance = this;
+            }
+            DontDestroyOnLoad(this);
+        }
+
+        void Start()
+        {
+            SceneManager.sceneLoaded += OnSceneLoaded;
+        }
+
+        public override void OnDestroy()
```

Comment:
does this need to be public?

Reply:
The event function it overrides in NetworkBehavior is public. The IDE gives me an error if I make it private, and if I just remove the override keyword I get a warning because it hides NetworkBehavior's method

Reply:
oh I forgot NetworkBehaviour implemented this. NVM nothing to see here